#!/bin/sh /usr/share/dpatch/dpatch-run
## 09-no-registration.dpatch by Frank Mehnert <frank@innotek.de>
##
## DP: Do not ask for registration. This is taken from SVN and can safely be
## DP: removed as soon as the files in SVN are realeased.

@DPATCH@

diff -Naurp virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk	2007-12-17 14:38:59.000000000 +0000
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/Makefile.kmk	2007-12-30 06:33:53.000000000 +0000
@@ -31,6 +31,9 @@ FORMS :=
 TRANSLATIONS :=
 IMAGES :=
 
+VBOX_WITH_REGISTRATION := 1
+VBOX_WITH_REGISTRATION_REQUEST := 1
+
 DEPTH = ../../../..
 include $(PATH_KBUILD)/header.kmk
 
@@ -38,10 +41,6 @@ include $(PATH_KBUILD)/header.kmk
 #
 # exclude inappropriate UI content
 #
-#ifndef VBOX_OSE
-VBOX_WITH_REGISTRATION := 1
-#endif
-
 ifndef VBOX_WITH_REGISTRATION
 VirtualBox_QT_UISRCS := $(filter-out ui/VBoxRegistrationDlg.ui,$(VirtualBox_QT_UISRCS))
 endif
@@ -217,6 +216,9 @@ endif
 ifdef VBOX_WITH_REGISTRATION
  VirtualBox_DEFS         += VBOX_WITH_REGISTRATION
 endif
+ifdef VBOX_WITH_REGISTRATION_REQUEST
+ VirtualBox_DEFS         += VBOX_WITH_REGISTRATION_REQUEST
+endif
 ifdef VBOX_WITH_ALSA
  VirtualBox_DEFS         += VBOX_WITH_ALSA
 endif
diff -Naurp virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp
--- virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp	2007-10-18 07:23:36.000000000 +0000
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp	2007-12-30 06:33:53.000000000 +0000
@@ -929,7 +929,9 @@ void VBoxConsoleWnd::finalizeOpenView()
     mIsOpenViewFinished = true;
     LogFlowFuncLeave();
 
+#ifdef VBOX_WITH_REGISTRATION_REQUEST
     vboxGlobal().showRegistrationDialog (false /* aForce */);
+#endif
 }
 
 /**
diff -Naurp virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/src/main.cpp virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/main.cpp
--- virtualbox-ose-1.5.4-dfsg.orig/src/VBox/Frontends/VirtualBox/src/main.cpp	2007-12-28 08:15:25.000000000 +0000
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/main.cpp	2007-12-30 06:33:53.000000000 +0000
@@ -242,7 +242,9 @@ int main (int argc, char **argv)
             {
                 a.setMainWidget (&vboxGlobal().selectorWnd());
                 vboxGlobal().selectorWnd().show();
+#ifdef VBOX_WITH_REGISTRATION_REQUEST
                 vboxGlobal().showRegistrationDialog (false /* aForce */);
+#endif
                 vboxGlobal().startEnumeratingMedia();
                 rc = a.exec();
             }
